compiler/rustc_span/src/def_id.rs

```diff
@@ -147,13 +147,17 @@ impl Borrow<Fingerprint> for DefPathHash {
 ///
 /// For more information on the possibility of hash collisions in rustc,
 /// see the discussion in [`DefId`].
-#[derive(Copy, Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
 #[derive(HashStable_Generic, Encodable, Decodable)]
-pub struct StableCrateId(pub(crate) u64);
+pub struct StableCrateId([u8; 8]);
```

Comment:
Does this deserve a `HalfFingerprint` type?

Reply:
I'm trying to create new types to represent 64-bit and 128-bit hashes that can provide some reasonable encapsulation to prevent slowly sliding back to where we are currently. It's not as easy as I hoped.